ReconstructOrder/workflow/multiDimProcess.py

```diff
@@ -401,6 +412,18 @@ def process_sample_imgs(img_io: Union[mManagerReader, PolAcquReader]=None,
             if img_io.bg_correct:
                 norm_sample = img_reconstructor.correct_background(norm_sample, stokes_bg)
 
+            if save_ret_Zavg:
+                stk_attribute_name = ['s0', 's1_norm', 's2_norm', 's3', 'polarization']
+                norm_sample_Zavg = StokesData()
+                [norm_sample_Zavg.s0,
+                 norm_sample_Zavg.s1_norm,
+                 norm_sample_Zavg.s2_norm,
+                 norm_sample_Zavg.s3,
+                 norm_sample_Zavg.polarization] = [np.mean(stack, axis=2) for stack in
+                                                  [norm_sample.__getattribute__(stack_name) for stack_name in stk_attribute_name]]
+
+                physical_data_Zavg = img_reconstructor.reconstruct_birefringence(norm_sample_Zavg)
+
```

Comment:
I think this can be moved to img_reconstructor so this already very long method doesn't get longer... in the next refactor PR if not in this PR.